app/controllers/markets_controller.rb

```diff
@@ -0,0 +1,66 @@
+class MarketsController < ApplicationController
+  def home
+    @markets = Market.order(:name)
+  end
+
+  def index
+  	user = params[:user_type]
+  	session[:user_type] = user unless user.nil?
+  	@buttons = true if session[:user_type] == 'market'
+  	@markets = Market.all if params[:order].nil?
```

Comment:
It seems like these three conditionals should be combined into one single conditional. All of these actions are mutually exclusive, and by writing them like this, you are implying that somehow each individual conditional could be true. This will cause the code to execute each conditional, but if the first one is true, the next two should not be executed.